pkg/components/nodejs/nodejs.go

```diff
@@ -0,0 +1,102 @@
+//go:build linux
+
+package nodejs
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/open-telemetry/opentelemetry-ebpf-instrumentation/pkg/beyla"
+	"github.com/open-telemetry/opentelemetry-ebpf-instrumentation/pkg/components/ebpf"
+	"github.com/open-telemetry/opentelemetry-ebpf-instrumentation/pkg/components/svc"
+)
+
+type NodeInjector struct {
+	log *slog.Logger
+	cfg *beyla.Config
+}
+
+// make sure not to use a variable name (such as that returned by
+// os.CreateTemp() to ensure that node does not load the file twice
+const fdExtractorPath = "/beyla_fdextractor.js"
+
+func NewNodeInjector(cfg *beyla.Config) *NodeInjector {
+	return &NodeInjector{
+		cfg: cfg,
+		log: slog.With("component", "nodejs.Injector"),
+	}
+}
+
+func writeFile(data []byte, path string) (cleanup func(), err error) {
+	file, err := os.Create(path)
+	if err != nil {
+		return nil, err
+	}
+
+	cleanup = func() {
+		file.Close()
+		os.Remove(file.Name())
+	}
+
+	if _, err := file.Write(data); err != nil {
+		cleanup()
+		return nil, err
+	}
+
+	if err := os.Chmod(file.Name(), 0o444); err != nil {
+		cleanup()
+		return nil, errors.New("failed to chmod file")
+	}
+
+	return cleanup, nil
+}
+
+func (i *NodeInjector) Enabled() bool {
```

Comment:
I think we only need this if traces are enabled? Correct? Metrics don't care about the internal context propagation.